amy/extforms/forms.py

```diff
@@ -199,20 +200,29 @@ def create_consent_field(self, term: Term) -> forms.ChoiceField:
     def validate_member_code(self, request: HttpRequest) -> dict:
         errors = dict()
         member_code = self.cleaned_data.get("member_code", "")
+        error_msg = (
+            "This code is invalid. "
+            "This may be due to a typo, an expired code, "
+            "or a code that has not yet been activated. "
+            "Please confirm that you have copied the code correctly, "
+            "or contact your site contact to verify your code."
+        )
 
         if not member_code:
             return
 
-        # ensure that code belongs to a membership
-        # TODO: implement further validation for active membership, etc
-        #       and update text accordingly
         try:
-            Membership.objects.get(registration_code=member_code)
+            # find relevant membership - may raise Membership.DoesNotExist
+            membership = Membership.objects.get(registration_code=member_code)
+            # confirm that membership is currently active
+            if (
+                not membership.agreement_start
+                <= date.today()
+                <= membership.agreement_end
```

Comment:
You could create a `Membership.active_on_date(d: date) -> bool` method for checking if membership is active on a given date. This would allow for some unit testing and fine-tuning in future if we wanted to extend the range the membership is considered active by a few days.